sdk/communication/azure-communication-chat/src/test/java/com/azure/communication/chat/ChatAsyncClientTest.java

```diff
@@ -57,92 +60,137 @@ protected void afterTest() {
 
     @Test
     public void canCreateThread() {
+        // Arrange
         CreateChatThreadOptions threadRequest = ChatOptionsProvider.createThreadOptions(
             firstThreadMember.getId(), secondThreadMember.getId());
 
-        ChatThreadAsyncClient chatThreadClient = client.createChatThread(threadRequest).block();
-        assertNotNull(chatThreadClient);
-        assertNotNull(chatThreadClient.getChatThreadId());
+        // Act & Assert
+        StepVerifier.create(client.createChatThread(threadRequest))
+            .assertNext(chatThreadClient -> {
+                assertNotNull(chatThreadClient);
+                assertNotNull(chatThreadClient.getChatThreadId());
+            })
+            .verifyComplete();
     }
 
     @Test
     public void canCreateThreadWithResponse() {
+        // Arrange
         CreateChatThreadOptions threadRequest = ChatOptionsProvider.createThreadOptions(
             firstThreadMember.getId(), secondThreadMember.getId());
 
-        ChatThreadAsyncClient chatThreadClient = client.createChatThreadWithResponse(threadRequest).block().getValue();
-        assertNotNull(chatThreadClient);
-        assertNotNull(chatThreadClient.getChatThreadId());
+        // Act & Assert
+        StepVerifier.create(client.createChatThreadWithResponse(threadRequest))
+            .assertNext(chatThreadClientResponse -> {
+                ChatThreadAsyncClient chatThreadClient = chatThreadClientResponse.getValue();
+                assertNotNull(chatThreadClient);
+                assertNotNull(chatThreadClient.getChatThreadId()); 
+            })
+            .verifyComplete();
     }
 
     @Test
     public void canGetChatThreadClient() {
+        // Arrange
         String threadId = "19:fe0a2f65a7834185b29164a7de57699c@thread.v2";
 
+        // Act
         ChatThreadAsyncClient chatThreadClient = client.getChatThreadClient(threadId);
+
+        // Assert
         assertNotNull(chatThreadClient);
         assertEquals(chatThreadClient.getChatThreadId(), threadId);
     }
 
     @Test
     public void canGetExistingChatThread() {
+        // Arrange
         CreateChatThreadOptions threadRequest = ChatOptionsProvider.createThreadOptions(
             firstThreadMember.getId(), secondThreadMember.getId());
         ChatThreadAsyncClient chatThreadClient = client.createChatThread(threadRequest).block();
```

Comment:
still need to remove this block. It's what I was asking Srikanta about during the tutorial. Let me find his answer about how to do it

Reply:
"What if we need to set up/create an object on the service first (as part of the arrange), before acting on it? Do we initialize the sync client too, and use it to arrange the setup? Or do we use a multi…"

you can chain multiple service calls asynchronously through Reactor operators like then().. so, the first call will do the necessary setup on the service and then call the next method that you want to test

Reply:
Done